### PR TITLE
Fix memory leak in isHostOpen

### PR DESCRIPTION
**Pull request by vitaliystoliarovcc**
Asana ticket: https://app.asana.com/0/1125859137800433/1200124046152476/f

Implemented request-response approach based on WS client/server.
Added support of attemps in request

### CODE REVIEW

File: lib/ws_clients/algos/index.js [hunk 1]
```diff
@@ -80,12 +76,7 @@ module.exports = class AlgosServerClient extends WSClient {
   }
 
   async isHostOpen (exID) {
-    return new Promise((resolve) => {
-      const reqID = `${nonce()}`
-      this.isOpenCallbacks[reqID] = resolve
-
-      this.send(['is.open', this.userID, exID, reqID])
-    })
+    return await this.request(['is.open', this.userID, exID])
```
Comment (avsek477) [inline lib/ws_clients/algos/index.js:79]: if we aren't using try catch block, return await is redundant.. you can just change this to <br /> `return this.request(['is.open', this.userID, exID])`

File: lib/ws_clients/algos/index.js [hunk 2]
```diff
@@ -16,7 +16,6 @@ const onIdentified = require('./handlers/on_identified')
 const onReloadedAO = require('./handlers/on_reloaded_ao')
```
Comment (avsek477) [inline lib/ws_clients/algos/index.js:16]: remove unused imports from this file

**Review by avsek477 — CHANGES_REQUESTED** [review]
lgtm other than the minor changes :)

File: lib/ws_clients/algos/index.js (hunk 1 above)
Comment (vitaliystoliarovcc) [reply lib/ws_clients/algos/index.js:79]: 'await' makes it clear to the developer that this.request is an asynchronous function without having to look at the implementation of that function

File: lib/ws_clients/algos/index.js (hunk 2 above)
Comment (vitaliystoliarovcc) [reply lib/ws_clients/algos/index.js:14]: done ✔️

**Review by robertkowalski — APPROVED** [review]
lgtm

**Pull request merged**